### PR TITLE
log queries and add SQL query options

### PR DESCRIPTION
**Pull request by nikkyandrei**
1) add option to log all queries if you set up SAVEQUERIES true in a config file (for example ) 
basically before you call this file you will get logs by  MysqliDb->queries

2) for MysqliDb->get | insert | update | delete
add after SELECT , INSERT , etc. possibility to add a string or an array of strings with query options like LOW_PRIORITY , QUICK, IGNORE, HIGH_PRIORITY

ex to check  http://dev.mysql.com/doc/refman/5.6/en/insert.html for
insert

I use this to create queries with priority on high loaded servers

future question add me on skype:nikkyandrei

### CODE REVIEW

File: MysqliDb.php [hunk 1]
```diff
@@ -11,6 +12,13 @@
  * @license   http://opensource.org/licenses/gpl-3.0.html GNU Public License
  * @version   2.0
  **/
+
```
Comment (avbdr) [inline MysqliDb.php]: can we get rid of this?

**Comment by avbdr** [comment]
I like the functionality of the patch a lot but can you please do the following:
1. Get rid of the defines mess on the top. 
2. Rename define name to MYSQLIDBTRACE or SAVEQUERIESSAVEQUERIES. I think first one is better due to its size.
3. Fix the code style. Right now I see lot of weird indentations and function names with '_' instead of a camelcase.
4. Add a section to a readme how to use new functionality.

Thank you

File: MysqliDb.php (hunk 1 above)
Comment (nikkyandrei) [inline MysqliDb.php]: there was that code.

**Comment by nikkyandrei** [comment]
1. ) & 2)  You have 2 defines.. SAVEQUERIES (i'm used with it from WP , and ideea is from WP core ,) and "MysqliDb_path_to_remove" = can be anything , I don't recomand to use something like ABSPATH(it is used a lot buy a lot of programers) but is exactly the value of ABSPATH = root directory, . if you want i can change it , come with a nice name what you want for both of them.  I recommand to use both of them on top with description to be more easy to use for developers that create custom frameworks or small aplications. 
"MysqliDb_path_to_remove" I like to use it this option to clean the path : ex 
path of file where is a query to execute : /home/user/public_html/core/inc/MyClassFolder/MyClassFile.php 
In logs i like to see only "core/inc/MyClassFolder/MyClassFile.php "
so i make: define('MysqliDb_path_to_remove', " /home/user/public_html/")


3) . I like more with "_" than CamelCase , but you can change it .

4.) I don't like to make descriptions  :( I have some problems with writing in English but i can provide you any information you want . 

Base Info :
1 to activate  track of all SQL queries you must define before loading the file of the class   define('SAVEQUERIES', true); this can be usualy added to a config file or some debug class that is loaded before the MysqliDb Class file 
after you define SAVEQUERIES as "true" you can see all queries that was loadedt until a point using: 
$db= new MysqliDb(.........);
$db->query($sql1);
$db->insert($sql2);
$db->update($sql3);

print_r($db->queries);

2 about Query options , to add to INSERT , SELECT , UPDATE, DELETE  query options like LOW_PRIORITY , QUICK, IGNORE, HIGH_PRIORITY 
simply you can do :
for INSERT LOW_PRIORITY  INTO `table` .......

$db->setQueryOption('LOW_PRIORITY ');
$db->insert($table,$param);

OR 

$db->setQueryOption('LOW_PRIORITY ')->insert($table,$param);

also you can use array for :

 INSERT LOW_PRIORITY  IGNORE INTO `table` .......

$db->setQueryOption(arrary('LOW_PRIORITY ','IGNORE'))->insert($table,$param);


PS  You are free to adapt , change an do anything you want with those ideas .
Also you can contact me directly on skype: nikkyandrei if you want more information / ideas

**Comment by avbdr** [comment]
sounds good. I will land the patch with my adjustments somewhere this week. 
Thanks for a patch.

**Comment by avbdr** [comment]
if someone have comments on the API leave them here

**Comment by avbdr** [comment]
Thank you. Merged with my changes https://github.com/joshcam/PHP-MySQLi-Database-Class/commit/fef36b1df5fbea61612f9c712eb913eaf9e262e6

**Pull request closed without merging**